api/controller/assignments.js

```diff
@@ -0,0 +1,904 @@
+        collection.countDocuments(query, function (err, count) {
+            var params = JSON.parse(JSON.stringify(req.query));
+            var route = req.route.path;
+            var options = getOptions(req, count, "+buddy_name");
+            //find all entries whic match with assignment id using aggregate
+            collection.aggregate([
+                {
+                    $match: {
+
+                        "content.metadata.assignmentId": assignmentId
+                    }
+                },
+                {
+                    $project: {
+                        _id: 0,
+                        content: {
+                            $filter: {
+                                input: "$content",
+                                as: "item",
+                                cond: { $eq: ["$$item.metadata.assignmentId", assignmentId] },
+                            }
+                        },
+                    }
+                },
+                {
+                    $sort: {
+                        "content.metadata.buddy_name": 1
+                    }
+                }
+
+            ]).toArray(function (err, items) {
+
+                if (err) {
+                    return res.status(500).send({
+                        error: "An error has occurred",
+                        code: 10
+                    });
+                } else {
+                    var data = {
+                        'deliveries': items,
+                        'offset': options.skip,
+                        'limit': options.limit,
+                        'total': options.total,
+                        'sort': options.sort[0][0] + "(" + options.sort[0][1] + ")",
+                        'links': {
+                            prev_page: (options.skip - options.limit >= 0) ? formPaginatedUrl(route, params, options.skip - options.limit, options.limit) : undefined,
+                            next_page: (options.skip + options.limit < options.total) ? formPaginatedUrl(route, params, options.skip + options.limit, options.limit) : undefined
+                        }
+                    };
+                    res.send(data);
+                }
+            });
+        });
+    });
+};
+
+//find assignment by id
+exports.findById = function (req, res) {
+    var assignmentId = req.params.assignmentId;
+    //validate
+    if (!mongo.ObjectID.isValid(assignmentId)) {
+        return res.status(401).send({
+            'error': "Invalid assignment id",
+            'code': 35
+        });
+    }
+    db.collection(assignmentCollection, function (err, collection) {
+        collection.findOne({ _id: new mongo.ObjectID(assignmentId) }, function (err, assignment) {
+            if (err) {
+                return res.status(500).send({
+                    'error': "An error has occurred",
+                    'code': 10
+                });
+            }
+            if (!assignment) {
+                return res.status(401).send({});
```

Comment:
Can you send an error and code with the error response?

Reply:
Also, why is status code 401 returned if assignment does not exists? This code is used for unauthorized requests.